recipes/matio/all/conanfile.py

```diff
@@ -54,7 +54,7 @@ def layout(self):
 
     def requirements(self):
         if self.options.with_hdf5:
-            self.requires("hdf5/1.14.1")
+            self.requires("hdf5/[>=1.14.0 <1.15]")
```

Comment:
Hi @tbeu, 
Thanks a lot for the contribution. We are currently introducing version ranges in Conan Center little by little and for the moment we only allow them for openssl, cmake and libcurl. Please check: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

Reply:
Yes, the zlib ranges are being introduced by the Conan team  to avoid conflicts in consumers as much as we can. There's a note in the docs:
![image](https://github.com/conan-io/conan-center-index/assets/5045666/c683fe33-837e-46ac-8350-37938e298bae)